packages/utils/src/embeddingWorker.ts

```diff
@@ -0,0 +1,13 @@
+import { cos_sim } from "@xenova/transformers";
+
+onmessage = async function (message) {
+  const { items, sourceEmbedding } = message.data;
+  const itemsWithSimilarity = await Promise.all(
+    items.map(async (item) => {
+      const score = await cos_sim(sourceEmbedding, item.embedding);
+      return { ...item, score };
+    })
+  );
+  const filteredItems = itemsWithSimilarity.filter((item) => item.score > 0.2);
+  postMessage(filteredItems);
+};
```

Comment:
Don't think we need a worker just for calculating the distance - can call that right from the UI thread.